tests/test-naughty-notification.lua

```diff
@@ -0,0 +1,30 @@
+local runner = require("_runner")
+local naughty = require("naughty")
+
+local steps = {}
+
+local n
+
+-- Regression test for #3625:
+-- Make sure that setting `timeout = 0` on an existing notification
+-- correctly stops and removes the timer.
+table.insert(steps, function()
+    n = naughty.notification({ text = "regression #3625", timeout = 1 })
+    n.timeout = 0
+    return true
+end)
+
+-- Wait for the notification's timeout to pass.
+table.insert(steps, function()
+    os.execute("sleep 2")
```

Comment:
Instead of sleeping 2 full second, can it usleep for 0.3s, check, then return `nil` if the condition isn't met? I have been bitten by hardcoded blocking `sleep` in tests before

Reply:
Not really. Technically, the test is for "will this notification exist indefinitely". And "infinity has passed" is not really a condition I can check for. The closes I have is "we're still there after the previous timeout value", but I need to know that said timeout has passed for that.

I guess I could try to add up elapsed time, but I don't have enough experience with the test framework's weird "emulate async by repeatedly calling the same step" to know if that's any better of an idea than sleeping for multiple seconds.

Reply:
Our test handle this internally. If something returns `nil` too many time, the test fails. This is transparent. Just check for the condition and return `nil` if it's not met, the framework contains the boilerplate and timeout/retry logic.

Reply:
As I said, I cannot "check for the condition". The condition is "infinity has passed and the notification is still visible".

I could access the notification's internal values to check `timer == nil`, but that wouldn't be an integration test.